src/components/LoadingIcon/PLoadingIcon.vue

```diff
@@ -1,17 +1,42 @@
 <template>
-  <div class="p-loading-icon">
+  <div class="p-loading-icon" :class="iconSizeClass">
     <div class="p-loading-icon__segment" />
     <div class="p-loading-icon__segment" />
     <div class="p-loading-icon__segment" />
     <div class="p-loading-icon__segment" />
   </div>
 </template>
 
+<script setup lang="ts">
+  import { computed, defineProps } from 'vue'
+
+  const props = withDefaults(defineProps<{
+    size?: 'small' | 'default' | 'large',
+  }>(), {
+    size: 'default',
+  })
+
+  const iconSizeClass = computed(() => `p-loading-icon--${props.size}`)
+</script>
+
 <style>
 .p-loading-icon { @apply
   relative
-  w-4
-  h-4
+}
+
+.p-loading-icon--default { @apply
+  w-3
+  h-3
+}
+
+.p-loading-icon--small { @apply
+  w-3
+  h-3
+}
```

Comment:
Why have default if its the same as small? Could just default the size prop to `small` yeah? Maybe this is copy pasted from another component, so don't take this question personally lol.